python/dgl/graphbolt/feature_store.py

```diff
@@ -134,3 +138,81 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
                 f"but got {ids.shape[0]} and {value.shape[0]}."
             )
             self._tensor[ids] = value
+
+
+def load_feature_stores(root_dir: str, feat_desc: dict, in_memory: bool):
```

Comment:
The metadata in canonical data format should have the information per feature store e.g.
edge_feature: {
  "a", "do", "b": {
    "format": "npy"
    "path": "blabla"
    "in_memory": no
  }
  "a", "did", "c": {
    "format": "csv"
    "path": "blabla"
    "in_memory": yes
  }
}

@Rhett-Ying

Reply:
For this load_feature_stores, the input should just be the metadata.

Reply:
As discussed offline, `gb.Dataset` will pass in a metadata class which has `format`, `path`, `in_memory` attributes.
```
class OnDiskFeatureData(pydantic_yaml.YamlModel):
    format: Enum[numpy]
    path: str
    in_memory: bool = False
```